### ISSUE TITLE
Loss of UDP access & thus revert to DERP after a RTM_DELROUTE

### ISSUE THREAD

**Issue by bradfitz** [tailscale/tailscale#643]
Observed in both 0.100 and 1.0.3 (per logs in #641).

LogID b395d6f472c00cd0e6ee49d4a016351043d4b7d1ab74bd0fd2e865b0c531c884

Things are fine for a bit, and then a netlink message 25 (RTM_DELROUTE, which we don't parse or handle) and we treat it like a major rebind-worthy link change event.

But after we do so, UDP is gone.

@steeve reports running Plex with its NAT punching too. Not sure how its NAT stuff works, or how it might break Tailscale.

```
2020-08-04 05:56:09.898849707 +0200 +0200: 6.8M/26.2M monitor: unhandled netlink msg type 0x756e6b6e6f776e28323529: {Length:60 Type:unknown(25) Flags:0 Sequence:0 PID:0}, "\x02 \x00\x00\xff\x02\xfe\x02\x00\x00\x00\x00\b\x00\x0f\x00\xff\x00\x00\x00\b\x00\x01\x00\xc0\xa8V\xa5\b\x00\a\x00\xc0\xa8V\xa5\b\x00\x04\x0
0\x02\x00\x00\x00"
2020-08-04 05:56:09.899482432 +0200 +0200: 6.8M/26.2M LinkChange(isExpensive=false); needsRebind=true
2020-08-04 05:56:09.899586161 +0200 +0200: 6.8M/26.2M magicsock: link change, binding new port
2020-08-04 05:56:09.90023421 +0200 +0200: 6.8M/26.2M magicsock: closing connection to derp-4 (rebind), age 0s
2020-08-04 05:56:09.900422635 +0200 +0200: 6.8M/26.2M magicsock: 0 active derp conns
2020-08-04 05:56:09.900581755 +0200 +0200: 6.8M/26.2M magicsock: ReSTUN: endpoint update active, need another later ("link-change-major")
2020-08-04 05:56:09.900864032 +0200 +0200: 6.9M/26.2M magicsock: adding connection to derp-4 for home-keep-alive
2020-08-04 05:56:09.901021929 +0200 +0200: 6.8M/26.2M magicsock: 1 active derp conns: derp-4=cr0s,wr0s
2020-08-04 05:56:09.901291185 +0200 +0200: 6.8M/26.2M derphttp.Client.Recv: connecting to derp-4 (fra)
...
2020-08-04 05:56:10.508002765 +0200 +0200: 6.9M/26.2M monitor: unhandled netlink msg type 0x756e6b6e6f776e28323529: {Length:60 Type:unknown(25) Flags:0 Sequence:0 PID:0}, "\x02\x18\x00\x00\xfe\x02\xfd\x01\x00\x00\x00\x00\b\x00\x0f\x00\xfe\x00\x00\x00\b\x00\x01\x00\xc0\xa8V\x00\b\x00\a\x00\xc0\xa8V\xa5\b\x00\x04\x00\x02\x00\x00\x00"
2020-08-04 05:56:11.799693591 +0200 +0200: 6.5M/26.2M netcheck: udp=false v4=false v6=false mapvarydest= hair=false portmap=? derp=0
2020-08-04 05:56:11.799947668 +0200 +0200: 6.5M/26.2M magicsock: netInfo update: NetInfo{varies= hairpin=false ipv6=false udp=false derp=#4 portmap=? link=""}
2020-08-04 05:56:11.800304825 +0200 +0200: 6.5M/26.2M control: NetInfo: NetInfo{varies= hairpin=false ipv6=false udp=false derp=#4 portmap=? link=""}
2020-08-04 05:56:11.800473077 +0200 +0200: 6.5M/26.2M control: cancelMapSafely: synced=false
2020-08-04 05:56:11.800636421 +0200 +0200: 6.5M/26.2M control: cancelMapSafely: wrote to channel
2020-08-04 05:56:11.801170543 +0200 +0200: 6.5M/26.2M magicsock: endpoints changed: 192.168.86.165:57900 (localAddresses)
2020-08-04 05:56:11.801962329 +0200 +0200: 6.5M/26.2M peer keys: [Yrcfe] [O9m9F]
2020-08-04 05:56:11.802079833 +0200 +0200: 6.5M/26.2M v0.100.0-153-gc6c838785 peers: 10039858/169572990 68160/41576
2020-08-04 05:56:11.80219369 +0200 +0200: 6.5M/26.2M control: client.newEndpoints(0, [192.168.86.165:57900])
2020-08-04 05:56:11.802343342 +0200 +0200: 6.5M/26.2M control: cancelMapSafely: synced=false
2020-08-04 05:56:11.80244209 +0200 +0200: 6.5M/26.2M control: cancelMapSafely: channel was full
2020-08-04 05:56:11.802775559 +0200 +0200: 6.5M/26.2M magicsock: starting endpoint update (link-change-minor)
2020-08-04 05:56:11.805300318 +0200 +0200: 6.7M/26.2M netcheck: probePortMapServices: me 192.168.86.165 -> gw 192.168.86.1
2020-08-04 05:56:11.944013933 +0200 +0200: 7.6M/26.2M control: Hostinfo: &{0.100.0-153-gc6c838785  b395d6f472c00cd0e6ee49d4a016351043d4b7d1ab74bd0fd2e865b0c531c884 linux nas [] [] [{[] tcp 22 sshd} {[] tcp 80 haproxy} {[] tcp 139 smbd} {[] tcp 443 haproxy} {[] tcp 445 smbd} {[] tcp 8887 haproxy} {[] tcp 8889 haproxy} {[] tcp 9091 transmission-da} {[] tcp 18811 StandaloneRpcSe} {[] tcp 51413 transmission-da} {[] tcp 53117 rally_service}] NetInfo{varies= hairpin=false ipv6=false udp=false derp=#4 portmap=? link=""}}
```

In all these logs, `udp=false`, but immediately prior to the rebind, UDP was working fine.

/cc @danderson for theories.

**Comment by crawshaw** [comment]
Is that with verbose netcheck logging? I'm curious if something went wrong with the probe planning, and no new UDP probes were sent (so UDP is never set to true).

**Comment by steeve** [comment]
It happened again this morning (France time):
```
2020/08/06 11:04:02 6.2M/21.6M magicsock: starting endpoint update (periodic)
2020/08/06 11:04:02 6.4M/21.6M netcheck: probePortMapServices: me 192.168.86.165 -> gw 192.168.86.1
2020/08/06 11:04:02 6.4M/21.6M netcheck: udp=true v6=false mapvarydest=false hair=false portmap=UMC v4a=81.67.224.126:45891 derp=4 derpdist=1v4:82ms,2v4:157ms,4v4:27ms
2020/08/06 11:04:26 6.8M/21.6M magicsock: starting endpoint update (periodic)
2020/08/06 11:04:26 6.2M/21.6M netcheck: probePortMapServices: me 192.168.86.165 -> gw 192.168.86.1
2020/08/06 11:04:26 6.2M/21.6M netcheck: udp=true v6=false mapvarydest=false hair=false portmap=UMC v4a=81.67.224.126:45891 derp=4 derpdist=1v4:83ms,2v4:157ms,4v4:28ms
2020/08/06 11:04:46 7.4M/21.6M magicsock: starting endpoint update (periodic)
2020/08/06 11:04:46 7.6M/21.6M netcheck: probePortMapServices: me 192.168.86.165 -> gw 192.168.86.1
2020/08/06 11:04:46 7.6M/21.6M netcheck: udp=true v6=false mapvarydest=false hair=false portmap=UMC v4a=81.67.224.126:45891 derp=4 derpdist=1v4:82ms,2v4:158ms,4v4:28ms
2020/08/06 11:04:47 7.7M/21.6M monitor: unhandled netlink msg type 0x756e6b6e6f776e28323529: {Length:60 Type:unknown(25) Flags:0 Sequence:0 PID:0}, "\x02 \x00\x00\xff\x02\xfd\x03\x00\x00\x00\x00\b\x00\x0f\x00\xff\x00\x00\x00\b\x00\x01\x00\xc0\xa8V\xff\b\x00\a\x00\xc0\xa8V\xa5\b\x00\x04\x00\x02\x00\x00\x00"
2020/08/06 11:04:47 7.8M/21.6M LinkChange(isExpensive=false); needsRebind=true
2020/08/06 11:04:47 7.7M/21.6M magicsock: link change, binding new port
2020/08/06 11:04:47 7.7M/21.6M magicsock: closing connection to derp-4 (rebind), age 11h8m46s
2020/08/06 11:04:47 7.7M/21.6M magicsock: 0 active derp conns
2020/08/06 11:04:47 7.8M/21.6M monitor: unhandled netlink msg type 0x756e6b6e6f776e28323529: {Length:60 Type:unknown(25) Flags:0 Sequence:0 PID:0}, "\x02 \x00\x00\xff\x02\xfd\x03\x00\x00\x00\x00\b\x00\x0f\x00\xff\x00\x00\x00\b\x00\x01\x00\xc0\xa8V\x00\b\x00\a\x00\xc0\xa8V\xa5\b\x00\x04\x00\x02\x00\x00\x00"
2020/08/06 11:04:47 7.7M/21.6M magicsock: starting endpoint update (link-change-major)
2020/08/06 11:04:47 6.2M/21.6M magicsock: adding connection to derp-4 for home-keep-alive
2020/08/06 11:04:47 6.3M/21.6M magicsock: 1 active derp conns: derp-4=cr0s,wr0s
2020/08/06 11:04:47 6.3M/21.6M derphttp.Client.Recv: connecting to derp-4 (fra)
2020/08/06 11:04:48 6.3M/21.6M LinkChange(isExpensive=false); needsRebind=true
2020/08/06 11:04:48 6.4M/21.6M magicsock: link change, binding new port
2020/08/06 11:04:48 6.4M/21.6M magicsock: closing connection to derp-4 (rebind), age 0s
2020/08/06 11:04:48 6.4M/21.6M magicsock: 0 active derp conns
2020/08/06 11:04:48 6.4M/21.6M magicsock: ReSTUN: endpoint update active, need another later ("link-change-major")
2020/08/06 11:04:48 6.4M/21.6M magicsock: adding connection to derp-4 for home-keep-alive
2020/08/06 11:04:48 6.4M/21.6M magicsock: 1 active derp conns: derp-4=cr0s,wr0s
2020/08/06 11:04:48 6.4M/21.6M derphttp.Client.Recv: connecting to derp-4 (fra)
2020/08/06 11:04:48 6.4M/21.6M monitor: unhandled netlink msg type 0x756e6b6e6f776e28323529: {Length:60 Type:unknown(25) Flags:0 Sequence:0 PID:0}, "\x02 \x00\x00\xff\x02\xfe\x02\x00\x00\x00\x00\b\x00\x0f\x00\xff\x00\x00\x00\b\x00\x01\x00\xc0\xa8V\xa5\b\x00\a\x00\xc0\xa8V\xa5\b\x00\x04\x00\x02\x00\x00\x00"
2020/08/06 11:04:48 6.4M/21.6M LinkChange(isExpensive=false); needsRebind=false
2020/08/06 11:04:48 6.4M/21.6M magicsock: ReSTUN: endpoint update active, need another later ("link-change-minor")
2020/08/06 11:04:48 6.4M/21.6M LinkChange(isExpensive=false); needsRebind=false
2020/08/06 11:04:48 6.5M/21.6M LinkChange(isExpensive=false); needsRebind=false
2020/08/06 11:04:48 7.4M/21.6M [RATE LIMITED] LinkChange(isExpensive=%v); needsRebind=%v
2020/08/06 11:04:48 6.3M/21.6M monitor: unhandled netlink msg type 0x756e6b6e6f776e28323529: {Length:60 Type:unknown(25) Flags:0 Sequence:0 PID:0}, "\x02\x18\x00\x00\xfe\x02\xfd\x01\x00\x00\x00\x00\b\x00\x0f\x00\xfe\x00\x00\x00\b\x00\x01\x00\xc0\xa8V\x00\b\x00\a\x00\xc0\xa8V\xa5\b\x00\x04\x00\x02\x00\x00\x00"
2020/08/06 11:04:49 6.6M/21.6M netcheck: udp=false v4=false v6=false mapvarydest= hair=false portmap=? derp=0
2020/08/06 11:04:49 6.6M/21.6M magicsock: netInfo update: NetInfo{varies= hairpin=false ipv6=false udp=false derp=#4 portmap=? link=""}
2020/08/06 11:04:49 6.6M/21.6M control: NetInfo: NetInfo{varies= hairpin=false ipv6=false udp=false derp=#4 portmap=? link=""}
2020/08/06 11:04:49 6.6M/21.6M control: cancelMapSafely: synced=true
```

**Comment by bradfitz** [comment]
If you update to the latest unstable builds (https://pkgs.tailscale.com/unstable/) there will be more logging at least. Try that and report back?

**Comment by steeve** [comment]
Back with more logs, on the unstable build:
```
2020/08/07 11:04:41 6.5M/34.5M netcheck: udp=true v6=false mapvarydest=false hair=false portmap=UMC v4a=81.67.224.126:42347 derp=4 derpdist=1v4:81ms,2v4:154ms,4v4:27ms
2020/08/07 11:04:52 6.3M/34.5M monitor: RTM_DELROUTE: {Family:2 DstLength:32 SrcLength:0 Tos:0 Table:255 Protocol:2 Scope:253 Type:3 Flags:0 Attributes:{Dst:192.168.86.255 Src:192.168.86.165 Gateway:<nil> OutIface:2 Priority:0 Table:255 Expires:<nil>}}
2020/08/07 11:04:52 6.4M/34.5M LinkChange(isExpensive=false); needsRebind=true
2020/08/07 11:04:52 6.4M/34.5M magicsock: link change, binding new port
2020/08/07 11:04:52 6.4M/34.5M magicsock: closing connection to derp-4 (rebind), age 12h16m32s
2020/08/07 11:04:52 6.4M/34.5M magicsock: 0 active derp conns
2020/08/07 11:04:52 6.4M/34.5M magicsock: starting endpoint update (link-change-major)
2020/08/07 11:04:52 6.4M/34.5M monitor: RTM_DELROUTE: {Family:2 DstLength:32 SrcLength:0 Tos:0 Table:255 Protocol:2 Scope:253 Type:3 Flags:0 Attributes:{Dst:192.168.86.0 Src:192.168.86.165 Gateway:<nil> OutIface:2 Priority:0 Table:255 Expires:<nil>}}
2020/08/07 11:04:52 6.4M/34.5M magicsock: adding connection to derp-4 for home-keep-alive
2020/08/07 11:04:52 6.4M/34.5M magicsock: 1 active derp conns: derp-4=cr0s,wr0s
2020/08/07 11:04:52 6.4M/34.5M derphttp.Client.Recv: connecting to derp-4 (fra)
2020/08/07 11:04:52 7.5M/34.5M LinkChange(isExpensive=false); needsRebind=true
2020/08/07 11:04:52 7.5M/34.5M magicsock: link change, binding new port
2020/08/07 11:04:52 7.5M/34.5M magicsock: closing connection to derp-4 (rebind), age 0s
2020/08/07 11:04:52 7.5M/34.5M magicsock: 0 active derp conns
2020/08/07 11:04:52 7.5M/34.5M monitor: RTM_DELROUTE: {Family:2 DstLength:32 SrcLength:0 Tos:0 Table:255 Protocol:2 Scope:254 Type:2 Flags:0 Attributes:{Dst:192.168.86.165 Src:192.168.86.165 Gateway:<nil> OutIface:2 Priority:0 Table:255 Expires:<nil>}}
2020/08/07 11:04:52 7.5M/34.5M magicsock: ReSTUN: endpoint update active, need another later ("link-change-major")
2020/08/07 11:04:52 7.5M/34.5M magicsock: adding connection to derp-4 for home-keep-alive
2020/08/07 11:04:52 7.6M/34.5M magicsock: 1 active derp conns: derp-4=cr0s,wr0s
2020/08/07 11:04:52 7.6M/34.5M derphttp.Client.Recv: connecting to derp-4 (fra)
2020/08/07 11:04:52 7.6M/34.5M LinkChange(isExpensive=false); needsRebind=false
2020/08/07 11:04:52 6.4M/34.5M magicsock: ReSTUN: endpoint update active, need another later ("link-change-minor")
2020/08/07 11:04:52 6.4M/34.5M LinkChange(isExpensive=false); needsRebind=false
2020/08/07 11:04:52 6.4M/34.5M LinkChange(isExpensive=false); needsRebind=false
2020/08/07 11:04:52 6.5M/34.5M [RATE LIMITED] LinkChange(isExpensive=%v); needsRebind=%v
2020/08/07 11:04:52 6.6M/34.5M monitor: RTM_DELROUTE: {Family:2 DstLength:24 SrcLength:0 Tos:0 Table:254 Protocol:2 Scope:253 Type:1 Flags:0 Attributes:{Dst:192.168.86.0 Src:192.168.86.165 Gateway:<nil> OutIface:2 Priority:0 Table:254 Expires:<nil>}}
2020/08/07 11:04:54 6.5M/34.5M netcheck: udp=false v4=false v6=false mapvarydest= hair=false portmap=? derp=0
2020/08/07 11:04:54 6.5M/34.5M magicsock: netInfo update: NetInfo{varies= hairpin=false ipv6=false udp=false derp=#4 portmap=? link=""}
2020/08/07 11:04:54 6.6M/34.5M control: NetInfo: NetInfo{varies= hairpin=false ipv6=false udp=false derp=#4 portmap=? link=""}
2020/08/07 11:04:54 6.6M/34.5M control: cancelMapSafely: synced=true
2020/08/07 11:04:54 6.5M/34.5M control: PollNetMap: context canceled
2020/08/07 11:04:54 6.6M/34.5M control: mapRoutine: state:authenticated
2020/08/07 11:04:54 6.6M/34.5M control: PollNetMap: stream=-1 :0 [81.67.224.126:42347 192.168.86.165:42347]
2020/08/07 11:04:54 6.5M/34.5M magicsock: endpoints changed: 192.168.86.165:38216 (localAddresses)
2020/08/07 11:04:54 6.8M/34.5M peer keys: [Yrcfe]
2020/08/07 11:04:54 6.9M/34.5M v1.1.57-g5062fd52b peers: 3776218/146548468
2020/08/07 11:04:54 6.9M/34.5M control: client.newEndpoints(0, [192.168.86.165:38216])
2020/08/07 11:04:54 6.9M/34.5M control: cancelMapSafely: synced=false
2020/08/07 11:04:54 6.9M/34.5M control: cancelMapSafely: wrote to channel
2020/08/07 11:04:54 6.9M/34.5M magicsock: starting endpoint update (link-change-minor)
2020/08/07 11:04:54 6.7M/34.5M netcheck: probePortMapServices: me 192.168.86.165 -> gw 192.168.86.1
2020/08/07 11:04:54 7.2M/34.5M control: new network map[0]:
2020/08/07 11:04:54 7.2M/34.5M control: mapRoutine: new map request during PollNetMap. canceling.
2020/08/07 11:04:54 7.2M/34.5M control: PollNetMap: context canceled
2020/08/07 11:04:54 7.2M/34.5M control: mapRoutine: state:authenticated
2020/08/07 11:04:54 7.2M/34.5M control: PollNetMap: stream=-1 :0 [192.168.86.165:38216]
2020/08/07 11:04:54 6.8M/34.5M control: mapRoutine: netmap received: state:synchronized
2020/08/07 11:04:54 6.8M/34.5M control: sendStatus: mapRoutine2: state:synchronized
2020/08/07 11:04:54 6.8M/34.5M netmap diff: (none)
2020/08/07 11:04:54 6.8M/34.5M magicsock: got updated network map; 6 peers (6 with discokey)
2020/08/07 11:04:56 6.5M/34.5M netcheck: udp=false v4=false v6=false mapvarydest= hair=false portmap=UMC derp=0
2020/08/07 11:04:56 6.5M/34.5M magicsock: netInfo update: NetInfo{varies= hairpin=false ipv6=false udp=false derp=#4 portmap=UMC link=""}
2020/08/07 11:04:56 6.5M/34.5M control: NetInfo: NetInfo{varies= hairpin=false ipv6=false udp=false derp=#4 portmap=UMC link=""}
2020/08/07 11:04:56 6.5M/34.5M control: cancelMapSafely: synced=true
2020/08/07 11:04:56 6.5M/34.5M control: PollNetMap: context canceled
2020/08/07 11:04:56 6.6M/34.5M control: mapRoutine: state:authenticated
2020/08/07 11:04:56 6.6M/34.5M control: PollNetMap: stream=-1 :0 [192.168.86.165:38216]
2020/08/07 11:04:56 6.7M/34.5M control: mapRoutine: netmap received: state:synchronized
2020/08/07 11:04:56 6.7M/34.5M control: sendStatus: mapRoutine2: state:synchronized
2020/08/07 11:04:56 6.7M/34.5M netmap diff: (none)
```

**Comment by steeve** [comment]
Interesting is that is happened on the same time as yesterday. Perhaps some maintenance running on the NAS?

NAS is connected to a gigabit switch, which is itself connected to a Google Wifi puck, which is connected to internet.

**Comment by steeve** [comment]
Interesting is going via DERP is much faster somehow, it used to be in the ~300ms range:
```
PING 100.107.40.118 (100.107.40.118): 56 data bytes
64 bytes from 100.107.40.118: icmp_seq=0 ttl=64 time=70.850 ms
64 bytes from 100.107.40.118: icmp_seq=1 ttl=64 time=67.350 ms
64 bytes from 100.107.40.118: icmp_seq=2 ttl=64 time=61.324 ms
64 bytes from 100.107.40.118: icmp_seq=3 ttl=64 time=60.412 ms
64 bytes from 100.107.40.118: icmp_seq=4 ttl=64 time=71.187 ms
```

**Comment by bradfitz** [comment]
Plan per meeting:

* add DefaultInterfaceRoute to interfaces.State
* add more logging on all netlink messages
* log before/after interfaces.State in userspaceEngine.setLinkState

Maybe we can start with new logging to confirm our theory of interfaces.State missing things first, then do the first item if/when confirmed.

**Comment by bradfitz** [comment]
@steeve, could you try again with version 1.1.109 or higher? We've added some logging a minor change that should't fix it but might. But logging should help regardless.

Let us know when it happens again.

**Comment by steeve** [comment]
I'll be able to do this weekend. We are having a heatwave in Paris so I turned off the NAS. Thank you for following up!

**Comment by steeve** [comment]
Updated my NAS a few minutes ago. Will report back.

**Comment by steeve** [comment]
So far the NAS hasn't lost UDP yet:
```
2020/08/19 15:06:31 7.3M/24.0M monitor: RTM_DELROUTE: src=192.168.86.209/0, dst=192.168.86.255/32, gw=, outif=2, table=255
2020/08/19 15:06:31 7.5M/24.0M monitor: RTM_DELROUTE: src=192.168.86.209/0, dst=192.168.86.0/32, gw=, outif=2, table=255
2020/08/19 15:06:31 7.5M/24.0M monitor: RTM_DELROUTE: src=192.168.86.209/0, dst=192.168.86.209/32, gw=, outif=2, table=255
2020/08/19 15:06:33 6.8M/24.0M netcheck: udp=true v6=false mapvarydest= hair=false portmap=? v4a=77.195.184.204:37793 derp=1 derpdist=1v4:82ms
2020/08/19 15:06:33 6.8M/24.0M magicsock: netInfo update: NetInfo{varies= hairpin=false ipv6=false udp=true derp=#1 portmap=? link=""}
2020/08/19 15:06:33 6.8M/24.0M control: NetInfo: NetInfo{varies= hairpin=false ipv6=false udp=true derp=#1 portmap=? link=""}
2020/08/19 15:06:33 6.7M/24.0M netcheck: udp=true v6=false mapvarydest=false hair=false portmap=UMC v4a=77.195.184.204:37793 derp=4 derpdist=1v4:82ms,3v4:170ms,4v4:37ms
2020/08/19 15:06:33 6.7M/24.0M magicsock: netInfo update: NetInfo{varies=false hairpin=false ipv6=false udp=true derp=#4 portmap=UMC link=""}
2020/08/19 15:06:33 6.7M/24.0M control: NetInfo: NetInfo{varies=false hairpin=false ipv6=false udp=true derp=#4 portmap=UMC link=""}
2020/08/19 15:06:43 7.8M/24.0M netcheck: udp=true v6=false mapvarydest=false hair=false portmap=UMC v4a=77.195.184.204:37793 derp=4 derpdist=1v4:85ms,3v4:173ms,4v4:40ms```

**Comment by steeve** [comment]
Update, it did loose UDP after `RTM_DELROUTE`, but got it back:
```
2020/08/20 15:06:35 7.7M/24.0M monitor: RTM_DELROUTE: src=192.168.86.209/0, dst=192.168.86.255/32, gw=, outif=2, table=255
2020/08/20 15:06:35 7.8M/24.0M monitor: RTM_DELROUTE: src=192.168.86.209/0, dst=192.168.86.0/32, gw=, outif=2, table=255
2020/08/20 15:06:35 6.6M/24.0M LinkChange: major, rebinding. New state: &{InterfaceIPs:map[eth0:[192.168.86.209] lo:[127.0.0.1]] InterfaceUp:map[eth0:true lo:true] HaveV6Global:false HaveV4:true IsExpensive:false DefaultRouteInterface:}
2020/08/20 15:06:35 6.6M/24.0M magicsock: link change, binding new port
2020/08/20 15:06:35 6.6M/24.0M magicsock: closing connection to derp-8 (rebind), age 8h54m56s
2020/08/20 15:06:35 6.6M/24.0M magicsock: 0 active derp conns
2020/08/20 15:06:35 6.6M/24.0M magicsock: adding connection to derp-8 for home-keep-alive
2020/08/20 15:06:35 6.6M/24.0M magicsock: 1 active derp conns: derp-8=cr0s,wr0s
2020/08/20 15:06:35 6.6M/24.0M derphttp.Client.Recv: connecting to derp-8 (lhr)
2020/08/20 15:06:35 6.6M/24.0M magicsock: starting endpoint update (link-change-major)
2020/08/20 15:06:35 6.7M/24.0M monitor: RTM_DELROUTE: src=192.168.86.209/0, dst=192.168.86.209/32, gw=, outif=2, table=255
2020/08/20 15:06:35 6.7M/24.0M LinkChange: major, rebinding. New state: &{InterfaceIPs:map[eth0:[192.168.86.209] lo:[127.0.0.1]] InterfaceUp:map[eth0:true lo:true] HaveV6Global:false HaveV4:true IsExpensive:false DefaultRouteInterface:eth0}
2020/08/20 15:06:35 6.7M/24.0M magicsock: link change, binding new port
2020/08/20 15:06:35 6.7M/24.0M magicsock: closing connection to derp-8 (rebind), age 0s
2020/08/20 15:06:35 6.7M/24.0M magicsock: 0 active derp conns
2020/08/20 15:06:35 6.7M/24.0M magicsock: ReSTUN: endpoint update active, need another later ("link-change-major")
2020/08/20 15:06:35 6.7M/24.0M magicsock: adding connection to derp-8 for home-keep-alive
2020/08/20 15:06:35 6.7M/24.0M magicsock: 1 active derp conns: derp-8=cr0s,wr0s
2020/08/20 15:06:35 6.7M/24.0M derphttp.Client.Recv: connecting to derp-8 (lhr)
2020/08/20 15:06:35 6.8M/24.0M LinkChange: minor
2020/08/20 15:06:35 6.8M/24.0M magicsock: ReSTUN: endpoint update active, need another later ("link-change-minor")
2020/08/20 15:06:35 6.8M/24.0M monitor: RTM_NEWROUTE: src=192.168.86.209/0, dst=192.168.86.209/32, gw=, outif=2, table=255
2020/08/20 15:06:35 6.8M/24.0M LinkChange: minor
2020/08/20 15:06:35 6.9M/24.0M LinkChange: minor
2020/08/20 15:06:35 6.9M/24.0M monitor: RTM_NEWROUTE: src=192.168.86.209/0, dst=192.168.86.0/24, gw=, outif=2, table=254
2020/08/20 15:06:35 6.7M/24.0M [RATE LIMITED] monitor: %s: src=%v, dst=%v, gw=%v, outif=%v, table=%v
2020/08/20 15:06:35 6.7M/24.0M LinkChange: minor
2020/08/20 15:06:36 6.8M/24.0M LinkChange: minor
2020/08/20 15:06:36 6.8M/24.0M [RATE LIMITED] LinkChange: minor
2020/08/20 15:06:37 7.1M/24.0M netcheck: udp=false v4=false v6=false mapvarydest= hair=false portmap=? derp=0
2020/08/20 15:06:37 7.1M/24.0M magicsock: netInfo update: NetInfo{varies= hairpin=false ipv6=false udp=false derp=#8 portmap=? link=""}
2020/08/20 15:06:37 7.1M/24.0M control: NetInfo: NetInfo{varies= hairpin=false ipv6=false udp=false derp=#8 portmap=? link=""}
2020/08/20 15:06:37 7.1M/24.0M control: cancelMapSafely: synced=true
2020/08/20 15:06:37 7.1M/24.0M control: PollNetMap: context canceled
2020/08/20 15:06:37 7.1M/24.0M control: mapRoutine: state:authenticated
2020/08/20 15:06:37 7.1M/24.0M control: PollNetMap: stream=-1 :0 [77.195.184.204:41678 192.168.86.209:41678]
2020/08/20 15:06:37 7.1M/24.0M magicsock: endpoints changed: 192.168.86.209:48455 (localAddresses)
2020/08/20 15:06:37 7.1M/24.0M peer keys: [Yrcfe]
2020/08/20 15:06:37 7.1M/24.0M v1.1.109-g42090beaf peers: 193121/6072564
2020/08/20 15:06:37 7.1M/24.0M control: client.newEndpoints(0, [192.168.86.209:48455])
2020/08/20 15:06:37 7.1M/24.0M control: cancelMapSafely: synced=false
2020/08/20 15:06:37 7.1M/24.0M control: cancelMapSafely: wrote to channel
2020/08/20 15:06:37 7.1M/24.0M magicsock: starting endpoint update (link-change-minor)
2020/08/20 15:06:37 7.3M/24.0M netcheck: probePortMapServices: me 192.168.86.209 -> gw 192.168.86.1
2020/08/20 15:06:37 6.6M/24.0M control: new network map[0]:
2020/08/20 15:06:37 6.6M/24.0M control: mapRoutine: new map request during PollNetMap. canceling.
2020/08/20 15:06:37 6.6M/24.0M control: PollNetMap: context canceled
2020/08/20 15:06:37 6.6M/24.0M control: mapRoutine: state:authenticated
2020/08/20 15:06:37 6.6M/24.0M control: PollNetMap: stream=-1 :0 [192.168.86.209:48455]
2020/08/20 15:06:37 6.7M/24.0M control: mapRoutine: netmap received: state:synchronized
2020/08/20 15:06:37 6.7M/24.0M control: sendStatus: mapRoutine2: state:synchronized
2020/08/20 15:06:37 6.7M/24.0M netmap diff: (none)
2020/08/20 15:06:37 6.7M/24.0M magicsock: got updated network map; 4 peers (4 with discokey)
2020/08/20 15:06:37 6.7M/24.0M netcheck: udp=true v6=false mapvarydest=false hair=false portmap=UMC v4a=77.195.184.204:48455 derp=8 derpdist=1v4:82ms,2v4:151ms,4v4:33ms,8v4:23ms,9v4:127ms,10v4:149ms
2020/08/20 15:06:37 6.7M/24.0M magicsock: netInfo update: NetInfo{varies=false hairpin=false ipv6=false udp=true derp=#8 portmap=UMC link=""}
2020/08/20 15:06:37 6.7M/24.0M control: NetInfo: NetInfo{varies=false hairpin=false ipv6=false udp=true derp=#8 portmap=UMC link=""}
2020/08/20 15:06:37 6.8M/24.0M control: cancelMapSafely: synced=true
2020/08/20 15:06:37 6.8M/24.0M control: PollNetMap: context canceled
2020/08/20 15:06:37 6.8M/24.0M control: mapRoutine: state:authenticated
2020/08/20 15:06:37 6.8M/24.0M control: PollNetMap: stream=-1 :0 [192.168.86.209:48455]
2020/08/20 15:06:37 6.8M/24.0M magicsock: endpoints changed: 77.195.184.204:48455 (stun), 192.168.86.209:48455 (localAddresses)
2020/08/20 15:06:37 6.8M/24.0M v1.1.109-g42090beaf peers: 193121/6072564
2020/08/20 15:06:37 6.8M/24.0M control: client.newEndpoints(0, [77.195.184.204:48455 192.168.86.209:48455])
2020/08/20 15:06:37 6.8M/24.0M control: cancelMapSafely: synced=false
2020/08/20 15:06:37 6.8M/24.0M control: cancelMapSafely: wrote to channel
2020/08/20 15:06:37 6.9M/24.0M control: mapRoutine: new map request during PollNetMap. canceling.
2020/08/20 15:06:37 6.9M/24.0M control: PollNetMap: context canceled
2020/08/20 15:06:37 6.9M/24.0M control: mapRoutine: state:authenticated
2020/08/20 15:06:37 6.9M/24.0M control: PollNetMap: stream=-1 :0 [77.195.184.204:48455 192.168.86.209:48455]
2020/08/20 15:06:37 7.0M/24.0M control: mapRoutine: netmap received: state:synchronized
2020/08/20 15:06:37 7.0M/24.0M control: sendStatus: mapRoutine2: state:synchronized
2020/08/20 15:06:37 7.0M/24.0M netmap diff: (none)
```

**Comment by DentonGentry** [comment]
Notes from bug scrub: this was reported right at the time the netlink watching was going in, looks like routing table is wrong for a while after DELROUTE. We can come up with a way to check if this is still happening with current versions, if we see UDP=true then an RTM_DELROUTE then UDP=false.

**Comment by josharian** [comment]
Rebind causing a permanent loss of UDP could have been caused by one of several bugs fixed recently (but all since 1.6). Those fixes should be in an unstable release very soon, and then in 1.8 next week.

However, I see that in one of the later incarnations of the bug, the loss of UDP was transient. The logging discussed has gone in, so it'd be useful to know if this happens again.

**Comment by DentonGentry** [comment]
As commented in April, we think the permanent loss of UDP has been resolved and if the transient loss is still there it has more logging now.

Barring further updates, we'll expect to close this during the next bug scrub.
